x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/date_histogram.tsx

```diff
@@ -334,3 +347,138 @@ function restrictedInterval(aggregationRestrictions?: Partial<IndexPatternAggRes
+                  id="xpack.lens.indexPattern.dateHistogram.autoLongerExplanation"
+                  defaultMessage="The Lens editor chooses an automatic interval for you by dividing the selected time range by the 
+                  {targetBarSetting} Kibana advanced setting. The calculation tries to present “nice” time interval buckets. The maximum 
+                  number of bars is set by the {maxBarSetting} value."
+                  values={{
+                    maxBarSetting: <EuiCode>{UI_SETTINGS.HISTOGRAM_MAX_BARS}</EuiCode>,
+                    targetBarSetting: <EuiCode>{UI_SETTINGS.HISTOGRAM_BAR_TARGET}</EuiCode>,
+                  }}
+                />
+              </p>
+            ),
+            advancedDescription: (
+              <p>
+                {i18n.translate('xpack.lens.indexPattern.dateHistogram.autoAdvancedExplanation', {
+                  defaultMessage: 'The specific interval adopted follow the logic in this table:',
+                })}
+              </p>
+            ),
+          }}
+        />
+      </EuiText>
+      <EuiBasicTable
+        items={wrapMomentPrecision(() =>
+          search.aggs.boundsDescendingRaw.map(({ bound, interval }) => ({
+            bound:
+              typeof bound === 'number'
+                ? infiniteBound
+                : `${upToLabel} ${humanDurationFormatter.convert(bound)}`,
+            interval: humanDurationFormatter.convert(interval),
+          }))
+        )}
+        columns={[
+          {
+            field: 'bound',
+            name: i18n.translate('xpack.lens.indexPattern.dateHistogram.autoBoundHeader', {
+              defaultMessage: 'Target Interval measured',
+            }),
+          },
+          {
+            field: 'interval',
+            name: i18n.translate('xpack.lens.indexPattern.dateHistogram.autoIntervalHeader', {
+              defaultMessage: 'Interval used',
+            }),
+          },
+        ]}
+      />
+    </EuiPopover>
+  );
+};
+
+// Below 5 seconds the "humanize" call returns the "few seconds" sentence, which is not ok for ms
+// This special config rewrite makes it sure to have precision also for sub-seconds durations
+// ref: https://github.com/moment/moment/issues/348
+function wrapMomentPrecision<T>(callback: () => T): T {
+  // Save default values
+  const roundingDefault = moment.relativeTimeRounding();
+  const units = [{ unit: 'ss', value: 0 }];
+  const defaultValues = units.map(({ unit }) => moment.relativeTimeThreshold(unit) as number);
+
+  moment.relativeTimeRounding((t) => {
+    const DIGITS = 2;
+    return Math.round(t * Math.pow(10, DIGITS)) / Math.pow(10, DIGITS);
+  });
+  units.forEach(({ unit, value }) => moment.relativeTimeThreshold(unit, value));
+
+  // Execute the format/humanize call in the callback
+  const result = callback();
+
+  // restore all the default values now in moment to not break it
+  units.forEach(({ unit }, i) => moment.relativeTimeThreshold(unit, defaultValues[i]));
```

Comment:
If there's only one unit, why loop?

Reply:
It used to be multiple unit settings.
I've left the loop for now in case I need to tweak some other units: can refactor to single one if it's all good.

Reply:
Found some rounding errors, so added few more units to the list.